### PR TITLE
Adding some tests to verify the available Locales

### PR DESCRIPTION
**Pull request by vahidR**
Adding some tests to verify that available locales list is always populated, and the default locale is "en_GB"

### CODE REVIEW

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java [hunk 1]
```diff
@@ -160,4 +162,30 @@ public void shouldNotMatchInvalidMessagesPropertiesFilesWithCreateMessagesProper
...
+    @Test
+    public void shoudTheDefaultLocalBeEnglish() {
+        // Given
+        List<String> locals = LocaleUtils.getAvailableLocales();
+
+        // When
+        String defaultLocal = locals.get(0);
+
+        // Then
+        assertThat(defaultLocal, is(equalTo("en_GB")));
```
Comment (kingthorin) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: Could you ensure locale has an "e" throughout, so that there's no confusion down the road?

**Comment by vahidR** [comment]
yeah.. I'll fix it

**Comment by kingthorin** [comment]
Tks. Can you rebase interactive and fixup or squash into a single comit? 

https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

**Comment by vahidR** [comment]
Alright.. that was a bit new to me.
I did rebase interactive with the "fixup" command.

**Comment by kingthorin** [comment]
Hmmm something still isn't right. After you've fixed up there should only be one commit in the PR.

Are you using eGit or command line?

**Comment by vahidR** [comment]
I'm using command line. 
But let's not wasting time anymore on fixing this.
Would you please close/delete this PR? I will send another pull request with the right format...

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java [hunk 2]
```diff
@@ -160,4 +162,30 @@ public void shouldNotMatchInvalidMessagesPropertiesFilesWithCreateMessagesProper
...
+    @Test
+    public void shoudAvailableLocalesBeNonEmpty() {
+        // Given
+        List<String> locales = LocaleUtils.getAvailableLocales();
+
+        // When
+        int localesSize = locales.size();
+
+        // Then
+        assertTrue(localesSize > 0);
```
Comment (thc202) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: Could use `assertThat(locales, is(not(empty())));` instead.

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java [hunk 3]
```diff
@@ -160,4 +162,30 @@ public void shouldNotMatchInvalidMessagesPropertiesFilesWithCreateMessagesProper
...
+    }
+
+    @Test
+    public void shoudTheDefaultLocaleBeEnglish() {
```
Comment (thc202) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: Probably better to call it "shouldHaveEnglishAsFirstAvailableLocale" instead. What matters most is that it's the first locale, not if it's default or not (it's not).

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java [hunk 4]
```diff
@@ -160,4 +162,30 @@ public void shouldNotMatchInvalidMessagesPropertiesFilesWithCreateMessagesProper
...
+        List<String> locales = LocaleUtils.getAvailableLocales();
+
+        // When
+        String defaultLocale = locales.get(0);
+
+        // Then
+        assertThat(defaultLocale, is(equalTo("en_GB")));
+    }
+
+
```
Comment (thc202) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: This line could be removed.

**Comment by thc202** [comment]
@vahidR you can close the pull request any time, although it's better to fix this one instead of creating another pull request (for the review context and such things).

**Comment by thc202** [comment]
btw, we will gladly help with the interactive rebase.

**Comment by vahidR** [comment]
Hi,

Alright... I will modify the code, apply your comments, and fix this rebase thing :)

**Comment by vahidR** [comment]
Done.. I rebased them to a single commit!

**Comment by kingthorin** [comment]
Looks goodLeave a comment

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java [hunk 5]
```diff
@@ -160,4 +163,28 @@ public void shouldNotMatchInvalidMessagesPropertiesFilesWithCreateMessagesProper
...
         }
     }
+
+    @Test
+    public void shoudAvailableLocalesBeNonEmpty() {
+        // Given
+        List<String> locales = LocaleUtils.getAvailableLocales();
+
+        // When
+        int localesSize = locales.size();
```
Comment (thc202) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: This statement could be removed (and the "When" word moved to same comment as "Given").

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java [hunk 6]
```diff
@@ -160,4 +163,28 @@ public void shouldNotMatchInvalidMessagesPropertiesFilesWithCreateMessagesProper
...
+        assertThat(locales, is(not(empty())));
+    }
+
+    @Test
+    public void shouldHaveEnglishAsFirstAvailableLocale() {
+        // Given
+        List<String> locales = LocaleUtils.getAvailableLocales();
+
+        // When
+        String defaultLocale = locales.get(0);
```
Comment (thc202) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: Could the variable be renamed to `firstAvailableLocale`?

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java (hunk 5 above)
Comment (vahidR) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: Done..

File: test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java (hunk 6 above)
Comment (vahidR) [inline test/org/zaproxy/zap/utils/LocaleUtilsUnitTest.java]: Done..

**Comment by vahidR** [comment]
@thc202 The requested changes have been done accordingly..

**Comment by thc202** [comment]
Looks good to me, thank you!

**Comment by psiinon** [comment]
Thanks @vahidR - how would you like to be included in the credits?
https://github.com/zaproxy/zap-core-help/wiki/HelpCredits#zap-extended-team

**Comment by vahidR** [comment]
@psiinon Thanks for merging..
My name is: Vahid Rafiei (@vahid_r)

**Comment by thc202** [comment]
Done in zaproxy/zap-core-help@1109a0779c3e5c7e963e9056e4fc7f16069b6f9b. Thanks!

**Pull request merged**